tests/e2e/onboarding.test.ts

```diff
@@ -131,6 +140,134 @@ test('onboarding with a short code', async ({ page, getOnboardingData }) => {
 	await expect(page).toHaveURL(`/onboarding`)
 })
 
+test('onboarding with GitHub OAuth', async ({ page }) => {
+	const code = `${MOCK_CODE_GITHUB}_${crypto.randomUUID()}_PLAYWRIGHT_TEST`
+
+	await page.route(/\/auth\/github(?!\/callback)/, async (route, request) => {
+		const headers = { ...request.headers(), [MOCK_CODE_GITHUB_HEADER]: code }
+		await route.continue({ headers })
+	})
+
+	const ghUser = await insertGitHubUser(code)!
+
+	await page.goto('/signup')
+	await page.getByRole('button', { name: /signup with github/i }).click()
+
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+	expect(
+		page.getByText(new RegExp(`welcome aboard ${ghUser.primaryEmail}`, 'i')),
+	).toBeVisible()
+
+	// fields are pre-populated for the user
+	const usernameInput = page.getByRole('textbox', { name: /username/i })
+	expect(usernameInput).toHaveValue(
+		// @ts-ignore
+		normalizeUsername(ghUser.profile.login),
+	)
+	expect(page.getByRole('textbox', { name: /^name/i })).toHaveValue(
+		ghUser.profile.name,
+	)
+
+	// button currently in 'idle' (neutral) state and so has got no companion
+	const createAccountButton = page.getByRole('button', {
+		name: /create an account/i,
+	})
+	expect(createAccountButton.getByRole('status')).not.toBeVisible()
+	expect(createAccountButton.getByText('cross')).not.toBeAttached()
+
+	// attempt 1
+	// Username is a string of alphanums and underscores from 3 to 20 chars
+	// long, that will be lowercased upon sumbission. See: app/utils/user-validation.ts:
+	await usernameInput.fill('U$er_name') // $ is invalid char
+	await createAccountButton.click()
+
+	await expect(createAccountButton.getByRole('status')).toBeVisible()
+	expect(createAccountButton.getByText('cross')).toBeAttached()
+	await expect(
+		page.getByText(
+			/username can only include letters, numbers, and underscores/i,
+		),
+	).toBeVisible()
+	// but we also never checked that privacy consent box
+	await expect(
+		page.getByText(
+			/you must agree to the terms of service and privacy policy/i,
+		),
+	).toBeVisible()
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+
+	// attempt 2
+	await usernameInput.fill('')
+	await createAccountButton.click()
+	await expect(page.getByText(/username is required/i)).toBeVisible()
+	// the next assertion will fail...
+	// await expect(createAccountButton.getByText('cross')).toBeAttached()
```

Comment:
Is this commented assertion left here intentionally?

Reply:
not at all, removed